### PR TITLE
299 bind oseir model

### LINKED ISSUE #299: bind oseir model
Add bindings for the updated OSEIR model implemented in #297.

### PR DESCRIPTION
**Pull request by MaxBetzDLR**
## Merge Request - GuideLine Checklist 

**Guideline** to check code before resolve WIP and approval, respectively.
As many checkboxes as possible should be ticked.

### Checks by code author:
* [x] There is at least one issue associated with the pull request.
* [x] The branch follows the naming conventions as defined in the [git workflow](git-workflow).
* [x] New code adheres with the [coding guidelines](coding-guidelines)
* [x] Tests for new functionality has been added
* [x] A local test was succesful
* [x] There is appropriate **documentation** of your work. (use doxygen style comments)
* [ ] If new third party software is used, did you pay attention to its license? Please remember to add it to the wiki after successful merging.
* [ ] If new mathematical methods or epidemiological terms are used, has the glossary been updated ? Did you provide further documentation ?
 is present or referenced. Please provide your references.
* [ ] The following questions are addressed in the documentation*:  Developers (what did you do?, how can it be maintained?), For users (how to use your work?), For admins (how to install and configure your work?)
* For documentation: Please write or update the Readme in the current working directory!

### Checks by code reviewer(s):
* [ ] Is the code clean of development artifacts e.g., unnecessary comments, prints, ...
* [ ] The ticket goals for each associated issue are reached or problems are clearly addressed (i.e., a new issue was introduced).
* [ ] There are appropriate **unit tests** and they pass.
* [ ] The git history is clean and linearized for the merge request.
* [ ] Coverage report for new code is acceptable. 

Closes #299

### CODE REVIEW

File: pycode/examples/simulation/oseir_simple.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    # Run Simulation
+    result = simulate(0, days, dt, model)
+    print(result.get_last_value())
+
+
+if __name__ == "__main__":
+    arg_parser = argparse.ArgumentParser(
+        'secir_simple', 
+        description = 'Simple example demonstrating the setup and simulation of the OSEIR model.')
+    arg_parser.add_argument('-p', '--show_plot', action='store_const', const=True, default=False)
```
Comment (dabele) [inline pycode/examples/simulation/oseir_simple.py:77]: I don't think you need this here.

File: pycode/examples/simulation/oseir_simple.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+from memilio.simulation.oseir import InfectionState as State
+import numpy as np
+import argparse
+
+
+def run_secir_simulation(show_plot = True):
+    """
+    Runs the c++ secir model using one age group 
+    and plots the results
+    """
```
Comment (dabele) [inline pycode/examples/simulation/oseir_simple.py:31]: I think this is copy paste from secir. Seir doesn't have age groups, and this example doesn't plot anything.

File: pycode/examples/simulation/oseir_simple.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    # Define Comartment names
+    compartments = ['Susceptible', 'Exposed', 'Infected', 'Recovered']
```
Comment (dabele) [inline pycode/examples/simulation/oseir_simple.py:34]: unused

File: pycode/examples/simulation/oseir_simple.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    # Define population of age groups
+    populations = [83000]
+
+    days = 100  # number of days to simulate
+    dt = 0.1
+    num_compartments = len(compartments)
```
Comment (dabele) [inline pycode/examples/simulation/oseir_simple.py:40]: unused

File: pycode/memilio-simulation/memilio/simulation/oseir.cpp [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+#include "utils/custom_index_array.h"
+#include "utils/parameter_set.h"
+#include "compartments/simulation.h"
+#include "compartments/compartmentalmodel.h"
+#include "epidemiology/populations.h"
+#include "ode_seir/model.h"
+#include "ode_seir/infection_state.h"
+#include "Eigen/Core"
+#include "pybind11/stl_bind.h"
+#include <vector>
```
Comment (dabele) [inline pycode/memilio-simulation/memilio/simulation/oseir.cpp:32]: I don't think you are using either of these includes.

**Review by dabele — CHANGES_REQUESTED** [review]

File: pycode/memilio-simulation/memilio/simulation/utils/index.h [hunk 6]
```diff
@@ -21,6 +21,7 @@
 #define PYMIO_INDEX_H
 
 #include "memilio/utils/index.h"
+#include "memilio/utils/custom_index_array.h"
```
[the remark this answers is not in the document]
Comment (MaxBetzDLR) [reply pycode/memilio-simulation/memilio/simulation/utils/index.h:24]: IndexPosition is used for casting which is defined in custom_index_array.h, so i thought it is required

[the remark this answers is not in the document]
Comment (dabele) [reply pycode/memilio-simulation/memilio/simulation/utils/index.h:24]: Ok, I didn't see that. That's an issue in the c++ library then. It's also questionable to use something in a `details` namespace outside of the library itself. But you don't need to fix that right now.

**Review by dabele — APPROVED** [review]

**Pull request merged**